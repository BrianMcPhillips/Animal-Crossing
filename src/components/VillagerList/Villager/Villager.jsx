import React from 'react';
import PropTypes from 'prop-types';
import styles from './Villager.module.css';

const Villager = ({ name, image }) => (
  <figure className={styles.Villager}>
    <img src={image} alt={name} />
    <figcaption>
      {name}
    </figcaption>
  </figure>
);

Villager.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Villager;
