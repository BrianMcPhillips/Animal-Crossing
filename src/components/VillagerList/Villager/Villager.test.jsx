import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Villager from './Villager';

describe('Villager component', () => {
  afterEach(() => cleanup());
  it('renders a Villager', () => {
    const { asFragment } = render(<Villager 
      name="Test Villager"
      image="http://placeholder.com/villager"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
