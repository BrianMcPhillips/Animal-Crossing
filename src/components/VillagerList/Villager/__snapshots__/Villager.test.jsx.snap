// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Villager component renders a Villager 1`] = `
<DocumentFragment>
  <figure
    class="Villager"
  >
    <img
      alt="Test Villager"
      src="http://placeholder.com/villager"
    />
    <figcaption>
      Test Villager
    </figcaption>
  </figure>
</DocumentFragment>
`;
